### ISSUE TITLE
Friends-only Games

### ISSUE THREAD

**Issue by AndrewDavis** [PredatH0r/SteamServerBrowser#26]
Is there any way to filter out friends-only games from Left4Dead2 servers? If not, can you add it?

Thanks.

**Comment by AndrewDavis** [comment]
_Lebump_

**Comment by PredatH0r** [comment]
you can click on the "=" sign below the "Bdy" column label and change the filter condition to ">0". Then you only see servers with buddies on them.

**Comment by PredatH0r** [comment]
If you want to filter out servers that contain ANY number of friends, just set the filter in the Bdy column to "< 1" (which only shows servers that have no friends on them).

If you want to filter out servers where all players are on your friends list, click on "filter editor", switch to Text view and set the filter to "[Bdy] <> [Humans]"

**Comment by AndrewDavis** [comment]
Okay yeah we're not talking about the same thing. A friends-only game in L4D2 is a game that is supposed to only let friends of the player who started the game into the game. Anyone else will not join the game. That's what friends-only means. The L4D2 client abides by this rule and will only send you to public games. SteamServerBrowser overrides this, however, allowing you to join any game on any server listed (which includes friends-only games). I'm looking for a way to filter such games out in the browser, so that I only play in public games.

**Comment by PredatH0r** [comment]
I could not find any information on how to even identify which servers are "friends-only".
If any of the variables under "Server Details" or "Rules" would indicate such servers, you can right-click the variable, add it as a column to the main table and then set a filter for it. But I haven't found any variable that suggests to indicate "friends only".

**Comment by AndrewDavis** [comment]
Yeah, unfortunately I don't see it in any of these panels either. I guess my question then would be: how does L4D2 do it? Also, I can see that my friend has joined a friends-only game when I view my friends in L4D2 itself (if any friends are online, a new menu option pops up at the bottom that if you click, it will show you where your friends are at and let you join their games). So somehow there's a way of telling, both when fetching servers to join (L4D2 Play Online) and when viewing an individual server that your friend is in.

The main reason I ask is not for myself; I don't really care that much. But, when I join a game, sometimes with a friend, people will secretly get mad at you and then later lash out at you or try to kick you because you've joined their game that they thought was private (they set it to Friends-Only instead of Public). I tell them to host locally if they really want that, but of course this isn't going to solve the problem for everyone because there are so many people and so many games.
